`sphinx` update causing `make doc` failure in CI

We are seeing failures in the CI. I [reran it out master](https://github.com/Bears-R-Us/arkouda/actions/runs/5882788504/job/15983025731) which had previous passed. I think this is like due to a dependency update of `sphinx`

```
Extension error (furo):
Handler <function _html_page_context at 0x7fac1f17e430> for event 'html-page-context' threw an exception (exception: This documentation is not using `furo.css` as the stylesheet. If you have set `html_style` in your conf.py file, remove it.)
make[1]: *** [Makefile:20: html] Error 2
make: *** [Makefile:438: /__w/arkouda/arkouda//docs/index.html] Error 2
writing output... [  1%] CONTRIBUTING_LINK
make[1]: Leaving directory '/__w/arkouda/arkouda/pydoc'
Error: Process completed with exit code 2.
```

--- a/arkouda-env-dev.yml
+++ b/arkouda-env-dev.yml
@@ -24,7 +24,7 @@ dependencies:
   - pexpect
   - pytest>=6.0
   - pytest-env
-  - Sphinx>=5.1.1,<7.2.0
+  - Sphinx>=5.1.1
   - sphinx-argparse
   - sphinx-autoapi
   - typed-ast

--- a/pydoc/requirements.txt
+++ b/pydoc/requirements.txt
@@ -21,7 +21,7 @@ libidn2
 pexpect
 pytest>=6.0
 pytest-env
-Sphinx>=5.1.1,<7.2.0
+Sphinx>=5.1.1
 sphinx-argparse
 sphinx-autoapi
 myst-parser

--- a/pydoc/setup/REQUIREMENTS.md
+++ b/pydoc/setup/REQUIREMENTS.md
@@ -40,7 +40,7 @@ The dependencies listed here are only required if you will be doing development 
 - `pexpect`
 - `pytest>=6.0`
 - `pytest-env`
-- `Sphinx>=5.1.1,<7.2.0`
+- `Sphinx>=5.1.1`
 - `sphinx-argparse`
 - `sphinx-autoapi`
 - `furo`

--- a/setup.py
+++ b/setup.py
@@ -156,7 +156,7 @@ setup(
     # projects.
     extras_require={  # Optional
         'dev': ['pexpect', 'pytest>=6.0', 'pytest-env',
-                'Sphinx>=5.1.1,<7.2.0', 'sphinx-argparse', 'sphinx-autoapi',
+                'Sphinx>=5.1.1', 'sphinx-argparse', 'sphinx-autoapi',
                 'mypy>=0.931,<0.990', 'typed-ast', 'black', 'isort',
                 'flake8', 'furo', 'myst-parser', 'linkify-it-py'],
     },

